lsp-icons.el

```diff
@@ -0,0 +1,93 @@
+;;; lsp-icons.el --- LSP icons management -*- lexical-binding: t; -*-
+;;
+;; Copyright (C) 2020 emacs-lsp maintainers
+;;
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+;;
+;;; Commentary:
+;;
+;;  LSP icons management
+;;
+;;; Code:
+
+(defgroup lsp-icons nil
+  "LSP icons"
+  :group 'lsp-mode
+  :tag "Icons")
+
+(defcustom lsp-headerline-breadcrumb-icons-enable t
+  "If non-nil, icons support is enabled for headerline-breadcrumb."
+  :type 'boolean
+  :group 'lsp-icons)
+
+(declare-function all-the-icons-material "ext:all-the-icons" t t)
+(declare-function lsp-treemacs-symbol-icon "ext:lsp-treemacs" (kind))
+(declare-function lsp-treemacs-get-icon "ext:lsp-treemacs" (icon-name))
+
+(defun lsp-icons--enabled-for-feature (feature)
```

Comment:
the issue is that the features that will benefit from that are not part of lsp-mode project, so we cannot enumerate them here.

I propose the following structure(or at least using this as base).

1. Have a functions get-file-icon-ALL-THE-ICONS/treemacs/whatever.
2. Have a property lsp-FEATURE-file-icon-function which defaults to something that make sense. The same for symbols. 

The global killswitch will be in the default function. At the same time, one could configure a particular feature to use different set of icons(e. g. headerlien to use all the icons, dap-mode to use smt else).

WDYT?

Reply:
I don't even know why this function is needed. Isn't providing overridable functions sufficient?